### PR TITLE
Download sky_engine and sky_services from cloud storage

### PR DESCRIPTION
**Pull request by abarth**
Downloading these packages from cloud storage simplifies our deployment
story because we can upload to cloud storage automatically from the
buildbot.

This patch also switches the responsibility for downloading the engine
artifacts to update_engine.sh. Centralizing this responsibility ensures
that the packages and the binaries are always in sync.

### CODE REVIEW

**Comment by abarth** [comment]
@eseidelGoogle Almost entirely works.  Needs https://codereview.chromium.org/1713373002 to land though.

File: bin/cache/update_engine.sh [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    unzip -o -q "$ARTIFACTS_ZIP" -d "$PLATFORM_PATH"
+    rm -f -- "$ARTIFACTS_ZIP"
+  }
+
+  download_artifacts android-arm
+
+  case "$(uname -s)" in
+    Darwin)
+      download_artifacts darwin-x64
+      chmod a+x "$ENGINE_ARTIFACT_PATH/darwin-x64/sky_snapshot"
```
Comment (eseidelGoogle) [inline bin/cache/update_engine.sh]: Zip can't do this?

**Comment by eseidelGoogle** [comment]
Seems like this coudl be a dart script, since it necessarily happens after we download dart. :(

lgtm, but shell scripting makes babies cry.

File: bin/cache/update_engine.sh (hunk 1 above)
Comment (abarth) [inline bin/cache/update_engine.sh]: The short answer is no.  There are some caveats if you're using GNU zip, but in reality we should switch to using tar, which does more sensible things.  In any case, this is just keeping the same behavior we had previously.  We can improve it in future iterations.

**Comment by abarth** [comment]
> Seems like this coudl be a dart script, since it necessarily happens after we download dart. :(

Possibly, but I'd like to extend this in the future to include all the Dart packages we depend on so that we can have a hermetically sealed SDK...  We can also worry about that in the future.

**Comment by Hixie** [comment]
my reaction was similar to Eric's. I would have set this up so that we get dart from a shell script, then do everything else in dart, including getting dependencies, compiling flutter tools, calling pub get, etc. But running code wins, so whatever. 🙂

**Comment by abarth** [comment]
Ok.  I'm going to land this patch, but we can continue to iterate on how we structure this code.

**Pull request merged**